artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -170,15 +215,26 @@ public void sendMessage(Message message, RoutingContext context, List<MessageRef
       }
    }
 
-   public static void validateProtocolData(MessageReference ref, SimpleString snfAddress) {
+   public static void validateProtocolData(ReferenceIDSupplier referenceIDSupplier, MessageReference ref, SimpleString snfAddress) {
       if (ref.getProtocolData() == null && !ref.getMessage().getAddressSimpleString().equals(snfAddress)) {
-         setProtocolData(ref);
+         setProtocolData(referenceIDSupplier, ref);
       }
    }
 
-   private static void setProtocolData(MessageReference ref) {
+   private static void setProtocolData(ReferenceIDSupplier referenceIDSupplier, MessageReference ref) {
       Map<Symbol, Object> daMap = new HashMap<>();
       DeliveryAnnotations deliveryAnnotations = new DeliveryAnnotations(daMap);
+
+      String brokerID = referenceIDSupplier.getListID(ref);
+
+      // getListID will return null when the message was generated on this broker.
+      // on this case we do not send the brokerID, and the ControllerTarget will get the information from the link.
+      // this is just to safe a few bytes and some processing on the wire.
+      if (brokerID != null) {
+         // not sending the brokerID, will make the other side to get the brokerID from the remote link's property
+         daMap.put(BROKER_ID, brokerID);
+      }
```

Comment:
Great, I was constantly thinking of this when reviewing the delta last week to stop munging the mirror id into the store id.

Reply:
thanks...

I will mark this as resolved.. just as a way for me to manage the changes..